examples/Core/itkImageIterationBenchmark.cxx

```diff
@@ -0,0 +1,376 @@
+
+  auto       inputIt = inputRange.begin();
+  auto       outputIt = outputRange.begin();
+  const auto inputEnd = inputRange.end();
+
+  const unsigned int componentsPerPixel = inputPtr->GetNumberOfComponentsPerPixel();
+  while (inputIt != inputEnd)
+  {
+    const InputPixelType & inputPixel = *inputIt;
+    OutputPixelType        outputPixel{ *outputIt };
+    for (unsigned int k = 0; k < componentsPerPixel; ++k)
+    {
+      outputPixel[k] = static_cast<typename OutputPixelType::ValueType>(inputPixel[k]);
+    }
+    *outputIt = outputPixel;
+
+    ++inputIt;
+    ++outputIt;
+  }
+}
+
+
+// Method 2b: ImageRegionRange approach using NumericTraits::GetLength()
+template <typename TInputImage, typename TOutputImage>
+void
+CopyImageRegionRangeNumericTraits(const TInputImage * inputPtr, TOutputImage * outputPtr)
+{
+  using InputPixelType = typename TInputImage::PixelType;
+  using OutputPixelType = typename TOutputImage::PixelType;
+
+  const typename TOutputImage::RegionType outputRegion = outputPtr->GetRequestedRegion();
+  typename TInputImage::RegionType        inputRegion = outputRegion;
+
+  auto inputRange = itk::ImageRegionRange<const TInputImage>(*inputPtr, inputRegion);
+  auto outputRange = itk::ImageRegionRange<TOutputImage>(*outputPtr, outputRegion);
+
+  auto       inputIt = inputRange.begin();
+  auto       outputIt = outputRange.begin();
+  const auto inputEnd = inputRange.end();
+
+  const unsigned int componentsPerPixel = itk::NumericTraits<OutputPixelType>::GetLength(*outputIt);
+  while (inputIt != inputEnd)
+  {
+    const InputPixelType & inputPixel = *inputIt;
+    OutputPixelType        outputPixel{ *outputIt };
+    for (unsigned int k = 0; k < componentsPerPixel; ++k)
+    {
+      outputPixel[k] = static_cast<typename OutputPixelType::ValueType>(inputPixel[k]);
+    }
+    *outputIt = outputPixel;
+    ++inputIt;
+    ++outputIt;
+  }
+}
+
+
+// Method 2c: ImageRegionRange approach - BUGGY VERSION demonstrating incorrect pixel reuse
+// NOTE: This method has a BUG - outputPixel is initialized once and reused, causing incorrect results
+// when the pixel type holds a reference to the image buffer (modifies first pixel repeatedly)
+template <typename TInputImage, typename TOutputImage>
+void
+CopyImageRegionRangeNumericTraitsAsRange(const TInputImage * inputPtr, TOutputImage * outputPtr)
+{
+  using InputPixelType = typename TInputImage::PixelType;
+  using OutputPixelType = typename TOutputImage::PixelType;
+
+  const typename TOutputImage::RegionType outputRegion = outputPtr->GetRequestedRegion();
+  typename TInputImage::RegionType        inputRegion = outputRegion;
+
+  auto outputRange = itk::ImageRegionRange<TOutputImage>(*outputPtr, outputRegion);
+  auto outputIt = outputRange.begin();
+
+  const unsigned int componentsPerPixel = itk::NumericTraits<OutputPixelType>::GetLength(*outputIt);
+  OutputPixelType    outputPixel{ *outputIt };
```

Comment:
```suggestion
  OutputPixelType    outputPixelRef{ *outputIt };
  OutputPixelType    outputPixel = outputPixelRef;
```
Does this change solve the bug? I my local debugging, it seems to do so.

Reply:
My local fix attempt: https://github.com/dzenanz/ITKPerformanceBenchmarking/commit/c696a55fefa28f68886d2b26ea6c67f8e3aad8f2